v0/destinations/appcues/transform.js

```diff
@@ -0,0 +1,184 @@
+const get = require("get-value");
+const set = require("set-value");
+const {
+  EventType
+} = require("../../../constants");
+
+const {
+  defaultRequestConfig,
+  getFieldValueFromMessage,
+  getValueFromMessage,
+  removeUndefinedAndNullValues,
+  removeUndefinedAndNullAndEmptyValues,
+} = require("../../util");
+
+const {
+  ConfigCategory,
+  mappingConfig,
+  getEndpoint,
+} = require("./config");
+
+const {
+  utcTimeToEpoch,
+} = require("./utils");
+
+function buildResponse(message, payload, endpoint) {
+  const response = defaultRequestConfig();
+  response.endpoint = endpoint;
+  response.userId = message.userId || message.anonymousId;
+  response.body.JSON = removeUndefinedAndNullValues(payload);
+  return {
+    ...response,
+    headers: {
+      "Content-Type": "application/json",
+    }
+  };
+}
+
+function getIdentifyPayload(message) {
+  let payload = {};
+  if (message.messageId) {
+    payload.request_id = message.messageId;
+  }
+  const traits = getFieldValueFromMessage(message, "traits");
+  if (traits && Object.keys(traits).length > 0) {
+    payload.profile_update = traits;
+  }
+  return payload;
+}
+
+function processIdentify(message, destination) {
+  return buildResponse(
+    message,
+    getIdentifyPayload(message),
+    getEndpoint(destination.Config.accountId, message.userId)
+  );
+}
+
+function getTrackPayload(message, mappingJson) {
+  let eventJson = {};
+  
+  // iterate over the destKeys and set the value if present
+  Object.keys(mappingJson).forEach(destKey => {
+    let value = get(message, mappingJson[destKey]);
+    if (value) {
+      eventJson[destKey] = value;
+    }
+  });
+  return eventJson;
+}
+
+function processTrackEvent(message, destination, mappingJson) {
+  let requestJson = {};
+  if (message.messageId) {
+    requestJson.request_id = message.messageId;
+  }
+
+  let eventJson = getTrackPayload(message, mappingJson);
+  eventJson.timestamp = utcTimeToEpoch(eventJson.timestamp);
+  requestJson.events = [eventJson];
+
+  return buildResponse(
+    message,
+    requestJson,
+    getEndpoint(destination.Config.accountId, message.userId)
+  );
+
+}
+
+function getProfilePayload(message) {
+  let profileJson = {};
+  const category = ConfigCategory.PROFILE;
+  const mappingJson = mappingConfig[category.name];
+  Object.keys(mappingJson).forEach(destKey => {
+    if (typeof mappingJson[destKey] === "object") {
+      const { sourceKeys } = mappingJson[destKey];
+      const value = getValueFromMessage(message, sourceKeys);
+      if(!!value)
+      {
+      set(profileJson, destKey, value);
+      }
+    } else {
+      set(profileJson, destKey, get(message, mappingJson[destKey]));
+    }
+  });
```

Comment:
Same comment as the comment on the method `getTrackPayload`

Reply:
Applied !